src/main/java/com/rackspacecloud/metrics/ingestionservice/influxdb/providers/DevTestTenantRouteProvider.java

```diff
@@ -8,10 +8,14 @@
     private String tenantRoutingServiceUrl;
 
     @Override
-    public TenantRoutes getRoute(String tenantId, String measurement, RestTemplate restTemplate) {
+    public TenantRoutes getRoute(String tenantId, String measurement, RestTemplate restTemplate) throws Exception {
         String requestUrl = String.format("%s/%s/%s", tenantRoutingServiceUrl, tenantId, measurement);
 
-        //TODO: Work on any exception handling if restTemplate throws exception
-        return restTemplate.getForObject(requestUrl, TenantRoutes.class);
+        try {
+            return restTemplate.getForObject(requestUrl, TenantRoutes.class);
+        }
+        catch (Exception e) {
+            throw new Exception(String.format("Exception thrown for requestUrl [%s]", requestUrl), e);
```

Comment:
Throwing `Exception` is a reasonable quick fix but not good coding style https://stackify.com/common-mistakes-handling-java-exception/

Reply:
I would love to learn this. I will go through the link you mentioned.